app/src/org/odk/collect/android/views/HierarchyElementView.java

```diff
@@ -33,39 +35,23 @@
     public HierarchyElementView(Context context, HierarchyElement it) {
         super(context);
 
-        setColor(it.getColor());
+        RelativeLayout layout = (RelativeLayout) inflate(context, R.layout.hierarchy_element_view, null);
 
-        mIcon = new ImageView(context);
-        mIcon.setImageDrawable(it.getIcon());
-        mIcon.setId(1);
-        mIcon.setPadding(0, 0, dipToPx(4), 0);
-
-        addView(mIcon, new RelativeLayout.LayoutParams(LayoutParams.WRAP_CONTENT,
-                LayoutParams.WRAP_CONTENT));
+        layout.setBackgroundColor(it.getColor());
 
-        mPrimaryTextView = new TextView(context);
-        mPrimaryTextView.setTextAppearance(context, android.R.style.TextAppearance_Large);
+        mPrimaryTextView = ((TextView)layout.findViewById(R.id.hev_primary_text));
         mPrimaryTextView.setText(it.getPrimaryText());
-        mPrimaryTextView.setId(2);
-        mPrimaryTextView.setGravity(Gravity.CENTER_VERTICAL);
-        LayoutParams l =
-            new RelativeLayout.LayoutParams(LayoutParams.WRAP_CONTENT, LayoutParams.WRAP_CONTENT);
-        l.addRule(RelativeLayout.RIGHT_OF, mIcon.getId());
-        addView(mPrimaryTextView, l);
-
-        mSecondaryTextView = new TextView(context);
+        mSecondaryTextView = ((TextView)layout.findViewById(R.id.hev_secondary_text));
         mSecondaryTextView.setText(it.getSecondaryText());
-        mSecondaryTextView.setTextAppearance(context, android.R.style.TextAppearance_Small);
-        mSecondaryTextView.setGravity(Gravity.CENTER_VERTICAL);
-
-        LayoutParams lp =
-            new RelativeLayout.LayoutParams(LayoutParams.WRAP_CONTENT, LayoutParams.WRAP_CONTENT);
-        lp.addRule(RelativeLayout.BELOW, mPrimaryTextView.getId());
-        lp.addRule(RelativeLayout.RIGHT_OF, mIcon.getId());
-        addView(mSecondaryTextView, lp);
+        mIcon = ((ImageView)layout.findViewById(R.id.hev_icon));
+        mIcon.setImageDrawable(it.getIcon());
 
-        setPadding(dipToPx(8), dipToPx(4), dipToPx(8), dipToPx(8));
+        addView(layout);
 
+        if (BuildConfig.DEBUG) {
+            Log.i("HEVTYPE", "Type of HEV (" + hashCode() + ") is " + it.getType());
+            Log.i("HEVTYPE", "Icon of HEV (" + hashCode() + ") is " + (it.getIcon() == null ? "null" : it.getIcon().toString()));
```

Comment:
I'm generally not a fan of info logging like this. Do you feel that it adds value after you are done working on this part of the code?